### PR TITLE
copr: add vectorized truncate_int

### PR DESCRIPTION
**Pull request by loxp**
### What problem does this PR solve?

UCP #6758 

Problem Summary:

### What is changed and how it works?

What's Changed:

### Related changes

Tests 

- Unit test

Side effects

### Release note

### CODE REVIEW

File: components/tidb_query_vec_expr/src/impl_math.rs [hunk 1]
```diff
@@ -10,6 +10,28 @@ use tidb_query_datatype::codec::{self, Error};
 use tidb_query_datatype::expr::EvalContext;
 use tidb_query_shared_expr::rand::MySQLRng;
 
+const I64_TEN_POWS: [i64; 19] = [
```
Comment (sticnarf) [inline components/tidb_query_vec_expr/src/impl_math.rs:13]: Prefer to use `static` instead of `const` to avoid putting this lookup table on the stack.

Comment (sticnarf) [reply components/tidb_query_vec_expr/src/impl_math.rs:13]: Sorry you can ignore me. The compiler will cleverly determine whether to inline them.

**Comment by sticnarf** [comment]
PTAL @zhongzc

**Comment by zhongzc** [comment]
Almost the same with #7813. The preview PR didn't pick up the challenge. How should we handle this case? @sticnarf

**Comment by sticnarf** [comment]
@zhongzc Do you mean #6556? I realize that the implementation here may be not right. Currently TiDB uses `TruncateUint` signature unlike https://github.com/tikv/tikv/pull/3532.

**Comment by sticnarf** [comment]
@loxp Could you use `truncate_int_with_int` for `TruncateInt` and `truncate_int_with_uint` for `TruncateUint`?

**Comment by zhongzc** [comment]
> @zhongzc Do you mean #6556? I realize that the implementation here may be not right. Currently TiDB uses `TruncateUint` signature unlike the moment of #3532.

Get you. 
https://github.com/pingcap/tidb/blob/a1763c1b6eecf24635e21591bbffba3d3f276a2f/expression/builtin_math_vec.go#L783-L847
is a more appropriate reference. @loxp

**Comment by loxp** [comment]
> > @zhongzc Do you mean #6556? I realize that the implementation here may be not right. Currently TiDB uses `TruncateUint` signature unlike the moment of #3532.
> 
> Get you.
> https://github.com/pingcap/tidb/blob/a1763c1b6eecf24635e21591bbffba3d3f276a2f/expression/builtin_math_vec.go#L783-L847
> is a more appropriate reference. @loxp

Sorry, I don't quite understand. Is it need to rewrite `ScalarFuncSig::TruncateInt` first?

**Comment by sticnarf** [comment]
I think the truncate int implementation is good, but you don't need a mapper. Instead, you can handle the uint case via `TruncateUint`.

**Pull request merged**